### PR TITLE
Introduce persistence context

### PR DESCRIPTION
**Pull request by janboll**
Context holds the value of a file managed by a template collection.

This fixes a bug, where a collection updating a file multiple times using more then one template, could lead to missing updates. Reason was, that previously everytime the file was read from persistence, i.e. gitlab. Now the content is held in the cache of a context manager.

Fixes: [APPSRE-10033](https://issues.redhat.com/browse/APPSRE-10033)

### CODE REVIEW

File: reconcile/templating/renderer.py [hunk 1]
```diff
@@ -40,8 +40,7 @@
 
 QONTRACT_INTEGRATION = "template-renderer"
 
-# Renders templates again to detect drift after one day
-CACHE_TTL_MINUTES = 24 * 60
+CACHE_TTL_MINUTES = 12
```
Comment (hemslo) [inline reconcile/templating/renderer.py:43]: Should this be a setting config?

File: reconcile/templating/renderer.py [hunk 2]
```diff
@@ -107,6 +106,31 @@ def read(self, path: str) -> Optional[str]:
         return current
 
 
+class PersistenceContext(FilePersistence):
+    def __init__(self, persistence: FilePersistence, dry_run: bool) -> None:
```
Comment (hemslo) [inline reconcile/templating/renderer.py:110]: A little strange to both inherit `FilePersistence` and pass in another `FilePersistence`, maybe call it something like `TransactionPersistence` and add some docstring?

Comment (janboll) [reply reconcile/templating/renderer.py:110]: Yeah, `TransactionablePersistence` was my first take as well. I can rename it.

**Comment by janboll** [comment]
Going to iterate over this again, to replace gitlab persistence with local git clone persistence. Should speed up and fix the need for caching as well and remove the need for state management.

**Pull request closed without merging**